comms/dht/src/connectivity/mod.rs

```diff
@@ -338,48 +287,44 @@ impl DhtConnectivity {
             return Ok(());
         }
 
-        if self.is_managed(conn.peer_node_id()) {
-            debug!(
-                target: LOG_TARGET,
-                "Node {} connected that is already managed by DhtConnectivity",
-                conn.peer_node_id()
-            );
-            return Ok(());
-        }
+        // if self.is_managed(conn.peer_node_id()) {
+        //     debug!(
+        //         target: LOG_TARGET,
+        //         "Node {} connected that is already managed by DhtConnectivity",
+        //         conn.peer_node_id()
+        //     );
+        //     return Ok(());
+        // }
 
         let current_dist = conn.peer_node_id().distance(self.node_identity.node_id());
-        let neighbour_distance = self.get_neighbour_max_distance();
-        if current_dist < neighbour_distance {
+        let bucket = current_dist.get_bucket(self.config.num_network_buckets);
             debug!(
                 target: LOG_TARGET,
-                "Peer '{}' connected that is closer than any current neighbour. Adding to neighbours.",
-                conn.peer_node_id().short_str()
+                "Peer '{}' connected. Adding to peer bucket {}.",
+                conn.peer_node_id().short_str(),
+                bucket.2
+
             );
 
-            if let Some(node_id) = self.insert_neighbour(conn.peer_node_id().clone()) {
-                // If we kicked a neighbour out of our neighbour pool but the random pool is not full.
-                // Add the neighbour to the random pool, otherwise remove it
-                if self.random_pool.len() < self.config.num_random_nodes {
-                    debug!(
-                        target: LOG_TARGET,
-                        "Moving peer '{}' from neighbouring pool to random pool", node_id
-                    );
-                    self.random_pool.push(node_id);
-                } else {
-                    debug!(target: LOG_TARGET, "Removing peer '{}' from neighbouring pool", node_id);
+            let bucket_size = if bucket.2 == 0 {
```

Comment:
Not a fan of the tuple. What is .0, .1 or .2? Think you should make a struct to name these fields as its hard to read.